### PR TITLE
Fix #8554 - Add test for public service implementations that don't use dependency objects

### LINKED ISSUE #8554: Add test for public service implementations that don't use dependency objects
Any service that has a public (i.e. not in Internal namespace) implementation type should group service dependencies using the pattern we use throughout the stack. Add automated test that any new service of this type uses this pattern.

### PR DESCRIPTION
**Pull request by anpete**

### CODE REVIEW

File: test/EFCore.Tests/ApiConsistencyTestBase.cs [hunk 1]
```diff
@@ -21,6 +23,41 @@ public abstract class ApiConsistencyTestBase
...
+        [Fact]
+        public void Service_implementations_should_use_dependencies_parameter_object()
+        {
+            var serviceCollection = new ServiceCollection();
+
+            AddServices(serviceCollection);
+
+            var badServiceTypes
+                = (from sd in serviceCollection
+                   where sd.ServiceType.Namespace.StartsWith("Microsoft.Entity", StringComparison.Ordinal)
```
Comment (smitpatel) [inline test/EFCore.Tests/ApiConsistencyTestBase.cs]: Why not `MS.EFCore`?

File: test/EFCore.Design.Tests/ApiConsistencyTest.cs [hunk 2]
```diff
@@ -3,11 +3,17 @@
...
 using Microsoft.EntityFrameworkCore.Design;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.EntityFrameworkCore
 {
     public class ApiConsistencyTest : ApiConsistencyTestBase
     {
+        protected override void AddServices(ServiceCollection serviceCollection)
+        {
+            throw new System.NotImplementedException();
```
Comment (smitpatel) [inline test/EFCore.Design.Tests/ApiConsistencyTest.cs]: Why does this throw? How this test work?

Comment (smitpatel) [inline test/EFCore.Design.Tests/ApiConsistencyTest.cs]: Yup it failed.

```
Microsoft.EntityFrameworkCore.ApiConsistencyTestBase.Service_implementations_should_use_dependencies_parameter_object()

Failed   Microsoft.EntityFrameworkCore.ApiConsistencyTest.Service_implementations_should_use_dependencies_parameter_object

Error Message:

 System.NotImplementedException : The method or operation is not implemented.

Stack Trace:

   at Microsoft.EntityFrameworkCore.ApiConsistencyTest.AddServices(ServiceCollection serviceCollection) in C:\projects\entityframework\test\EFCore.Design.Tests\ApiConsistencyTest.cs:line 14

   at Microsoft.EntityFrameworkCore.ApiConsistencyTestBase.Service_implementations_should_use_dependencies_parameter_object() in C:\projects\entityframework\test\EFCore.Tests\ApiConsistencyTestBase.cs:line 34

[xUnit.net 00:00:12.2767815]   Finished:    Microsoft.EntityFrameworkCore.Design.Tests


Total tests: 232. Passed: 231. Failed: 1. Skipped: 0.

Test Run Failed.

```

Comment (anpete) [inline test/EFCore.Design.Tests/ApiConsistencyTest.cs]: This was a mistake, I have fixed it locally but not re-pushed yet.

**Comment by ErikEJ** [comment]
providers beware?

**Pull request merged**